### PR TITLE
Issue228 cleanup activity changes

### LINKED ISSUE #228: Cleanup Activity Changes
Redmorello has made a bulk of the changes but there are some items that need to be cleaned up. Changes are needed ASAP so going to make changes from this branch
https://github.com/aarondcoleman/Fitbit.NET/pull/198

### PR DESCRIPTION
**Pull request by dwaynefitabase**
FIxes #228 

An extension of the changes from Redmorello

### CODE REVIEW

File: Fitbit.Portable.Tests/FatTests.cs [hunk 1]
```diff
@@ -16,7 +16,7 @@ public class FatTests
     {
         public Fixture fixture { get; set; }
 
-        [TestFixtureSetUp]
+        [OneTimeSetUp]
```
Comment (WestDiscGolf) [inline Fitbit.Portable.Tests/FatTests.cs]: New one on me :)

File: Fitbit.Portable.Tests/Fitbit.Portable.Tests.csproj [hunk 2]
```diff
@@ -256,6 +261,34 @@
...
+    <None Include="SampleData\ActivityGoals.json" />
+    <None Include="SampleData\AddSubscriptionResponse.json" />
+    <None Include="SampleData\ApiError-Request-Forbidden.json" />
+    <None Include="SampleData\ApiError-Request-StaleToken.json" />
+    <None Include="SampleData\ApiError-Request-Unauthorized.json" />
+    <None Include="SampleData\ApiError-Request-BadRequest.json" />
+    <None Include="SampleData\ApiSubscriptionNotification.json" />
+    <None Include="SampleData\GetActivities.json" />
+    <None Include="SampleData\GetHeartRateTimeSeries.json">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
```
Comment (WestDiscGolf) [inline Fitbit.Portable.Tests/Fitbit.Portable.Tests.csproj]: Does this need to be different from the others?

File: Fitbit.Portable.Tests/Fitbit.Portable.Tests.csproj [hunk 3]
```diff
@@ -256,6 +261,34 @@
...
+    </None>
+    <None Include="SampleData\GetHeartRateIntradayTimeSeries.json" />
+    <None Include="SampleData\GetActivityLogsList.json">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
```
Comment (WestDiscGolf) [inline Fitbit.Portable.Tests/Fitbit.Portable.Tests.csproj]: Same as above? need to be different?

File: Fitbit.Portable/IFitbitClient.cs [hunk 4]
```diff
@@ -37,9 +35,12 @@ public interface IFitbitClient
         Task<WaterData> GetWaterAsync(DateTime date);
         Task<WaterLog> LogWaterAsync(DateTime date, WaterLog log);
         Task DeleteWaterLogAsync(long logId);
-
         Task<List<ApiSubscription>> GetSubscriptionsAsync();
         Task<ApiSubscription> AddSubscriptionAsync(APICollectionType apiCollectionType, string uniqueSubscriptionId, string subscriberId = default(string));
         Task DeleteSubscriptionAsync(APICollectionType collection, string uniqueSubscriptionId, string subscriberId = null);
+        Task<ActivityLog> LogActivityAsync(ActivityLog model);
+        Task<HeartActivitiesTimeSeries> GetHeartRateTimeSeries(DateTime date, DateRangePeriod dateRangePeriod, string userId = "-");
```
Comment (WestDiscGolf) [inline Fitbit.Portable/IFitbitClient.cs]: GetHeartRateTimeSeriesAsync to keep with the pattern

File: Fitbit.Portable/IFitbitClient.cs [hunk 5]
```diff
@@ -37,9 +35,12 @@ public interface IFitbitClient
...
+        Task<HeartActivitiesIntraday> GetHeartRateIntraday(DateTime date, HeartRateResolution resolution, string userId = "-");
```
Comment (WestDiscGolf) [inline Fitbit.Portable/IFitbitClient.cs]: GetHeartRateIntradayAsync to keep with the pattern

File: Fitbit.Portable/IFitbitClient.cs [hunk 6]
```diff
@@ -37,9 +35,12 @@ public interface IFitbitClient
...
+        Task<List<ActivityLogsList>> GetActivityLogsList(DateTime? beforeDate, DateTime? afterDate, int limit = 20, string encodedUserId = default(string));
```
Comment (WestDiscGolf) [inline Fitbit.Portable/IFitbitClient.cs]: GetActivityLogsListAsync to keep with the pattern

File: Fitbit.Portable/Models/ActivityLogsList.cs [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+
+    //public class HeartRateZone
+    //{
+    //    public int max { get; set; }
+    //    public int min { get; set; }
+    //    public int minutes { get; set; }
+    //    public string name { get; set; }
+    //}
+
+    public class ManualValuesSpecified
```
Comment (WestDiscGolf) [inline Fitbit.Portable/Models/ActivityLogsList.cs]: All classes should be in their own files

File: Fitbit.Portable/Models/HeartActivitiesIntraday.cs [hunk 8]
```diff
@@ -19,9 +19,25 @@ public class HeartActivitiesIntraday
     public class DatasetInterval
     {
         public DateTime Time { get; set; }
-        public int Value { get; set;}
+        public int Value { get; set; }
     }
 
+    public class IntradayActivitiesHeart
```
Comment (WestDiscGolf) [inline Fitbit.Portable/Models/HeartActivitiesIntraday.cs]: All classes should be in their own files :)

File: Fitbit.Portable/Models/HeartActivitiesIntradayTimeSeries.cs [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+namespace Fitbit.Api.Portable.Models
+{
+    public class HeartActivitiesIntradayTimeSeries
+    {
+        [JsonProperty(PropertyName = "activities-heart")]
+        public List<ActivitesHeart> ActivitiesHeart { get; set; }
+        [JsonProperty(PropertyName = "activities-heart-intraday")]
+        public ActivitesHeartIntraday ActivitiesHeartIntraday { get; set; }
+    }
+    public class ActivitesHeart
```
Comment (WestDiscGolf) [inline Fitbit.Portable/Models/HeartActivitiesIntradayTimeSeries.cs]: All classes should be in their own files

File: Fitbit.Portable/FitbitClient.cs [hunk 10]
```diff
@@ -964,6 +908,164 @@ public Task<TimeSeriesDataListInt> GetTimeSeriesIntAsync(TimeSeriesResourceType
...
+        /// </summary>
+        /// <param name="beforeDate">The date in the format yyyy-MM-ddTHH:mm:ss. Only yyyy-MM-dd is required. Either beforeDate or afterDate must be specified. Set sort to desc when using beforeDate.</param>
+        /// <param name="afterDate">The date in the format yyyy-MM-ddTHH:mm:ss. Only yyyy-MM-dd is required. Either beforeDate or afterDate must be specified. Set sort to asc when using afterDate.</param>
+        /// <param name="limit">The max of the number of entries returned (maximum: 20)</param>
+        /// <param name="encodedUserId">encoded user id, can be null for current logged in user</param>
+        /// <returns>ActivityLog</returns>
+        public async Task<List<ActivityLogsList>> GetActivityLogsList(DateTime? beforeDate, DateTime? afterDate, int limit = 20, string encodedUserId = default(string))
+        {
+            var apiCall = string.Empty;
+            limit = limit > 20 ? 20 : limit;
```
Comment (WestDiscGolf) [inline Fitbit.Portable/FitbitClient.cs]: Missing the check for negative values as per the original code review / pull request

File: Fitbit.Portable/FitbitClient.cs [hunk 11]
```diff
@@ -964,6 +908,164 @@ public Task<TimeSeriesDataListInt> GetTimeSeriesIntAsync(TimeSeriesResourceType
...
+
+            apiCall = FitbitClientHelperExtensions.ToFullUrl("/1/user/{0}/activities/list.json?{1}={2}&sort={3}&limit={4}&offset={5}", encodedUserId, dateString, date, sort, limit, offset);
+
+            HttpResponseMessage response = await HttpClient.GetAsync(apiCall);
+            await HandleResponse(response);
+            var responseBody = await response.Content.ReadAsStringAsync();
+            return (new JsonDotNetSerializer() { RootProperty = "activities" }).Deserialize<List<ActivityLogsList>>(responseBody);
+        }
+
+
```
Comment (WestDiscGolf) [inline Fitbit.Portable/FitbitClient.cs]: Extra whitespace not needed between methods

File: Fitbit.Portable/FitbitClient.cs [hunk 12]
```diff
@@ -964,6 +908,164 @@ public Task<TimeSeriesDataListInt> GetTimeSeriesIntAsync(TimeSeriesResourceType
...
+        /// <summary>
+        /// Requests the Heart Rate Time Series for a specific time period.
+        /// </summary>
+        /// <param name="date">The end date of the period specified.</param>
+        /// <param name="dateRangePeriod">The range for which data will be returned.</param>
+        /// <param name="userId">The encoded ID of the user.</param>
+        /// <returns></returns>
+        public async Task<HeartActivitiesTimeSeries> GetHeartRateTimeSeries(DateTime date, DateRangePeriod dateRangePeriod, string userId = "-")
+        {
+
```
Comment (WestDiscGolf) [inline Fitbit.Portable/FitbitClient.cs]: Extra whitespace not needed in methods

File: Fitbit.Portable.Tests/FatTests.cs (hunk 1 above)
Comment (dwaynefitabase) [reply Fitbit.Portable.Tests/FatTests.cs]: Our CI build agents didn't like `TestFixtureSetUp` used with V3 of the test framework ;-|

File: Fitbit.Portable/IFitbitClient.cs (hunk 4 above)
Comment (dwaynefitabase) [reply Fitbit.Portable/IFitbitClient.cs]: GetHeartRateTimeSeries is the name used in current versions. RedMorello updated it, but this is a braking change so we're gonna keep it the same for now. https://github.com/aarondcoleman/Fitbit.NET/blob/079c870529030d1775f08653569115eaaa6b67c8/Fitbit.Portable/FitbitClient.cs#L396

File: Fitbit.Portable/IFitbitClient.cs (hunk 5 above)
Comment (dwaynefitabase) [reply Fitbit.Portable/IFitbitClient.cs]: https://github.com/aarondcoleman/Fitbit.NET/blob/079c870529030d1775f08653569115eaaa6b67c8/Fitbit.Portable/FitbitClient.cs#L422

File: Fitbit.Portable/IFitbitClient.cs (hunk 6 above)
Comment (dwaynefitabase) [reply Fitbit.Portable/IFitbitClient.cs]: This one can be changed.

File: Fitbit.Portable/Models/HeartActivitiesIntradayTimeSeries.cs (hunk 9 above)
Comment (dwaynefitabase) [reply Fitbit.Portable/Models/HeartActivitiesIntradayTimeSeries.cs]: 👍

File: Fitbit.Portable/FitbitClient.cs (hunk 10 above)
Comment (dwaynefitabase) [reply Fitbit.Portable/FitbitClient.cs]: 👍

File: Fitbit.Portable/FitbitClient.cs (hunk 12 above)
Comment (dwaynefitabase) [reply Fitbit.Portable/FitbitClient.cs]: 👍

File: Fitbit.Portable/FitbitClient.cs (hunk 11 above)
Comment (dwaynefitabase) [reply Fitbit.Portable/FitbitClient.cs]: 👍

File: Fitbit.Portable/Models/ActivityLogsList.cs (hunk 7 above)
Comment (dwaynefitabase) [reply Fitbit.Portable/Models/ActivityLogsList.cs]: 👍

File: Fitbit.Portable/Models/HeartActivitiesIntraday.cs (hunk 8 above)
Comment (dwaynefitabase) [reply Fitbit.Portable/Models/HeartActivitiesIntraday.cs]: 👍

**Pull request merged**